src/scene/start/boot.scene.ts

```diff
@@ -36,6 +40,10 @@ export class BootScene extends Phaser.Scene {
 
   public update() {
     console.log('go to main')
-    this.scene.start('main')
+
+    const numComponentScenes = [];
+    const randomStart = randomInt(2, 5);
+    // this.scene.start(`component-${randomStart}`)
+    this.scene.start(`component-4`)
```

Comment:
Rooms scenes don't have access to share global state to know the randomized order. I was thinking of making a game state singleton that any class can access. it can get initialized in the boot scene